extension-localization/src/main/java/com/mapbox/maps/extension/localization/Localization.kt

```diff
@@ -19,8 +19,13 @@ import java.util.*
  * `["format",["coalesce",["get","name_zh-Hans"],["get","name"]],{}]`
  */
 internal fun setMapLanguage(locale: Locale, style: StyleInterface, layerIds: List<String>?) {
+  var convertedLocale = "name_${locale.language}"
+  if (!isSupportedLanguage(convertedLocale)) {
```

Comment:
How should user determine if locale is supported or not, do we have some public API for that?

Reply:
You can refer to https://docs.mapbox.com/vector-tiles/reference/mapbox-streets-v8/#common-fields and https://docs.mapbox.com/vector-tiles/reference/mapbox-streets-v7/#name-fields

Reply:
I see. Since we have those lists of supported languages in the code - could we add strong guarantee that only correct language is set by user? Create some locale enum / wrapper / etc or throw an exception to signal if incorrect locale is passed?

Reply:
fully agree with your remarks @yunikkk, created an internal ticket to make sure our next iteration of i18n provides a better developer experience. Considering we will be deprecating this API soon, I'm not sure if we should do anything right now.